### PR TITLE
FR 67 - Add additional daily resource limit toggles

### LINKED ISSUE #67: Suggestion: Add mob related configurability to extraction limits
- Currently, there is no individually toggable way to turn extraction limits off for mob-related drops.

I'd like the ability to have config/setting options to enable/disable certain parts of the resource extraction limits.

- For example, disabling the mob farming mechanic that disables non-interaction mob drops and keeping generalized block-related extraction limits in place.

### PR DESCRIPTION
**Pull request by Firav**
### Description

Adds additional toggable settings for daily extraction limits.
- blocks (enables/disables daily resource extraction limits for block related checks)
- drops (enables/disables daily resource extraction limits for entity drop related checks)
- shearing (enables/disables daily resource extraction limits for shearing related checks)
- fishing (enables/disables daily resource extraction limits for fishing related checks)

### Resolves
Closes #67

### CODE REVIEW

File: src/main/java/io/github/townyadvanced/townyresources/listeners/TownyResourcesBukkitEventListener.java [hunk 1]
```diff
@@ -44,15 +44,15 @@ public void onBlockBreak(BlockBreakEvent event) {
...
+		if(TownyResourcesSettings.isEnabled() && TownyResourcesSettings.areResourceExtractionLimitsEnabled() && TownyResourcesSettings.areShearingExtractionLimitsEnabled()) {
 			//Dispensers cannot shear entities
 			event.setCancelled(true);
 		}	
 	}
 	
 	@EventHandler(ignoreCancelled = true)
 	public void onPlayerShearEntityEvent(PlayerShearEntityEvent event) {
-		if(TownyResourcesSettings.isEnabled()) {
+		if(TownyResourcesSettings.isEnabled() && TownyResourcesSettings.areShearingExtractionLimitsEnabled()) {
```
Comment (LlmDl) [inline src/main/java/io/github/townyadvanced/townyresources/listeners/TownyResourcesBukkitEventListener.java:55]: This one should probably also have a `TownyResourcesSettings.areResourceExtractionLimitsEnabled()` added to it, I think its probably been missing it, no?

File: src/main/java/io/github/townyadvanced/townyresources/controllers/PlayerExtractionLimitsController.java [hunk 2]
```diff
@@ -78,9 +78,9 @@ public static void loadAllResourceExtractionCategories() throws Exception{
      * @param event the event
      */
     public static void processEntityDamageByEntityEvent(EntityDamageByEntityEvent event) {
-        if(!TownyResourcesSettings.areResourceExtractionLimitsEnabled())
+        if(!TownyResourcesSettings.areResourceExtractionLimitsEnabled() || !TownyResourcesSettings.areDropsExtractionLimitsEnabled())
             return;
-    
+
```
Comment (LlmDl) [inline src/main/java/io/github/townyadvanced/townyresources/controllers/PlayerExtractionLimitsController.java:83]: line 83 and line 106 had white spacing changes made which should be reverted.

**Review by LlmDl — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/townyadvanced/townyresources/listeners/TownyResourcesBukkitEventListener.java (hunk 1 above)
Comment (Firav) [reply src/main/java/io/github/townyadvanced/townyresources/listeners/TownyResourcesBukkitEventListener.java]: You're correct. Added in 4934d53.

File: src/main/java/io/github/townyadvanced/townyresources/controllers/PlayerExtractionLimitsController.java (hunk 2 above)
Comment (Firav) [reply src/main/java/io/github/townyadvanced/townyresources/controllers/PlayerExtractionLimitsController.java:83]: Formatting resolved in 022cee9

Comment (LlmDl) [reply src/main/java/io/github/townyadvanced/townyresources/controllers/PlayerExtractionLimitsController.java:83]: Line 83 is still off, line 87 has now been changed.

**Review by LlmDl — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/townyadvanced/townyresources/controllers/PlayerExtractionLimitsController.java (hunk 2 above)
Comment (Firav) [reply src/main/java/io/github/townyadvanced/townyresources/controllers/PlayerExtractionLimitsController.java]: Attempt number 2 haha 8312a3c

Comment (Firav) [reply src/main/java/io/github/townyadvanced/townyresources/controllers/PlayerExtractionLimitsController.java:83]: a1316cc

File: src/main/java/io/github/townyadvanced/townyresources/controllers/PlayerExtractionLimitsController.java [hunk 3]
```diff
@@ -78,13 +78,13 @@ public static void loadAllResourceExtractionCategories() throws Exception{
...
```
Comment (LlmDl) [inline src/main/java/io/github/townyadvanced/townyresources/controllers/PlayerExtractionLimitsController.java:83]:
```suggestion
    
```

File: src/main/java/io/github/townyadvanced/townyresources/controllers/PlayerExtractionLimitsController.java [hunk 4]
```diff
@@ -78,13 +78,13 @@ public static void loadAllResourceExtractionCategories() throws Exception{
...
         //Return if not a mob
         if(!(event.getEntity() instanceof Mob))
             return;
-                
+
```
Comment (LlmDl) [inline src/main/java/io/github/townyadvanced/townyresources/controllers/PlayerExtractionLimitsController.java:87]:
```suggestion
                
```

**Review by LlmDl — CHANGES_REQUESTED** [review]
Accept suggestions to revert whitespace changes

**Review by LlmDl — APPROVED** [review]

**Pull request merged**